extensions/big-data-cluster/src/bigDataCluster/dialog/intializingComponent.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Deferred } from '../../common/promise';
+
+export abstract class InitializingComponent {
```

Comment:
do you see this class maybe useful for others as well? candidate for azdata namespace?

Reply:
Hm, I'll think about that. Currently we're only really using it in the more complicated ModelView stuff like dashboards so it isn't too widely spread but yeah maybe useful for everyone to have easy access to it. I'll keep it in mind as we make more stuff whether we need it more commonly available.